server/pubEdgeProposal/api.js

```diff
@@ -0,0 +1,71 @@
+import fetch from 'node-fetch';
+
+import app, { wrap } from 'server/server';
+import { parseUrl } from 'utils/urls';
+import { isDoi } from 'utils/dois';
+
+import { getPubDataFromUrl } from './queries';
+
+const createPubEdgeProposalFromPub = (pub, url) => {
+	const { attributions, avatar, description, doi, lastPublishedAt, title } = pub;
+	const contributors = attributions.map((a) => a.name);
+
+	return {
+		targetPub: {
```

Comment:
⚠️ Don't cast `Pub` to the `ExternalPublication` shape — for the sake of consistency with the rest of the `<PubEdge>` callsites, we should provide a Pub with the associated models required here. Those turn out to be specified [by the `getOptionsForIncludedPub` helper function in my PR](https://github.com/pubpub/pubpub/blob/cc7507cfd561e5dcabcb9fbb16242bf4d6d1fa1f/server/utils/queryHelpers/edgeOptions.js#L10).

You could choose to copy this code, interline it, or wait for my PR to land to resolve this.

Reply:
I'll wait for your PR.